compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -1048,6 +1049,22 @@ fn trait_def(tcx: TyCtxt<'_>, def_id: DefId) -> ty::TraitDef {
     )
 }
 
+fn trait_alias_def(tcx: TyCtxt<'_>, def_id: DefId) -> ty::TraitAliasDef {
+    let item = tcx.hir().expect_item(def_id.expect_local());
+    let hir::ItemKind::TraitAlias(_, bounds) = item.kind else {
+        span_bug!(item.span, "trait_alias_def_of_item invoked on non-trait alias")
+    };
+
+    ty::TraitAliasDef {
+        def_id,
+        explicit_bounds: bounds
+            .iter()
+            .filter_map(hir::GenericBound::trait_ref)
+            .filter_map(hir::TraitRef::trait_def_id)
```

Comment:
This ignores cases where these methods return `None`.  Is that okay, or are there some edges here to be wary of?